app/models/spotlight/reindex_progress.rb

```diff
@@ -0,0 +1,101 @@
+module Spotlight
+  ##
+  # ReindexProgress is a class that models the progress of reindexing a list of resources
+  class ReindexProgress
+    MINUTES_IN_PROGRESS = 10
+
+    def initialize(resource_list)
+      @resources = if resource_list.present?
+                     resource_list
+                   else
+                     null_resources
+                   end
+    end
+
+    def in_progress?
+      return unless finished
+      finished > MINUTES_IN_PROGRESS.minutes.ago
```

Comment:
Yeah, I can do that.  Let me make that change. 

Also, I wonder since we have the `index_status` now if we can be more clever here.  Like possibly `resources.any?(&:waiting?) && finished > MINUTES_IN_PROGRESS.minutes.ago`.

Otherwise I think this will require one resource to have completed before the reindexing is reported as in progress.